### PR TITLE
Implemented student views for assignments

### PR DESCRIPTION
**Pull request by rmathis**
@machristie Please review, but I have not yet tested in an environment yet so I am marking this as a draft

### CODE REVIEW

**Comment by rmathis** [comment]
This encompasses stories 499, 518, and 495

**Review by machristie — COMMENTED** [review]
This is looking really good @rmathis . I left a couple comments, but don't see any major issues.

File: src/main/javascript/src/views/student/StudentQuiz.vue [hunk 1]
```diff
@@ -150,6 +264,61 @@ export default {
...
+    },
+    selectedSubmissionDateSubmitted() {
+      return moment(this.selectedSubmission?.dateSubmitted).format('MMMM Do YYYY hh:mm');
+    },
+    timeBeforeSubmission() {
+      const time = this.selectedSubmission?.dateSubmitted - this.selectedSubmission?.dateCreated;
+      return isNaN(time) ? '' : moment.duration(time, "milliseconds").humanize();
+    },
+    currentScore() {
+      return `${this.selectedSubmission?.totalAlteredGrade} / ${this.assignmentData?.maxPoints}`;
```
[the remark this answers is not in the document]
Comment (rmathis) [reply src/main/javascript/src/views/student/StudentQuiz.vue]: Fixed

File: src/main/javascript/src/views/student/StudentQuiz.vue [hunk 2]
```diff
@@ -150,6 +264,61 @@ export default {
...
+
+      return `${kept ? kept : 0} / ${this.assignmentData?.maxPoints}`;
+    },
+    muted() {
+      if (!this.assignmentData) { return true; }
+      const { allowStudentViewResponses, studentViewResponsesAfter, studentViewResponsesBefore } = this.assignmentData;
+      if (allowStudentViewResponses) {
+        const now = Date.now();
+        const isAfter = moment(now).isAfter(studentViewResponsesAfter);
+        const isBefore = moment(now).isBefore(studentViewResponsesBefore);
```
[the remark this answers is not in the document]
Comment (rmathis) [reply src/main/javascript/src/views/student/StudentQuiz.vue]: Fixed

File: src/main/javascript/src/views/student/StudentQuiz.vue [hunk 3]
```diff
@@ -275,7 +275,14 @@ export default {
       return isNaN(time) ? '' : moment.duration(time, "milliseconds").humanize();
     },
     currentScore() {
-      return `${this.selectedSubmission?.totalAlteredGrade} / ${this.assignmentData?.maxPoints}`;
+      let grade;
+      if (!this.selectedSubmission) {
+        grade = '-';
+      } else {
+        const { totalAlteredGrade, alteredCalculatedGrade } = this.selectedSubmission;
+        grade = totalAlteredGrade ? totalAlteredGrade : alteredCalculatedGrade;
```
Comment (machristie) [inline src/main/javascript/src/views/student/StudentQuiz.vue:283]: So, it's possible that totalAlteredGrade is 0 (or 0.0) and thus not null but also would evaluate to false here. I think here the code should check if totalAlteredGrade is not equal to null.

**Review by machristie — CHANGES_REQUESTED** [review]
This looks good @rmathis 👍 . I just have one small nit-picky change request.

File: src/main/javascript/src/views/student/StudentQuiz.vue (hunk 3 above)
Comment (rmathis) [reply src/main/javascript/src/views/student/StudentQuiz.vue]: Good catch, fixed

**Review by machristie — APPROVED** [review]
Looks great, thanks @rmathis !

**Pull request merged**